shared/common-adapters/tab-bar.native.js

```diff
@@ -29,6 +29,18 @@ class SimpleTabBarButton extends Component<void, ItemProps, void> {
   }
 }
 
+const BadgeComponent = ({badgeNumber, badgePosition}: {badgeNumber: number, badgePosition: TabBadgePosition}) => {
```

Comment:
why not just pass a badgeStyle into this thing so any component can push the badges around. The badge position in this common component likely won't work for any other types of tab bars